packages/tsutils/src/PDFExporter.ts

```diff
@@ -0,0 +1,63 @@
+import cookie from 'cookie';
+import puppeteer from 'puppeteer';
+
+type RequestBody = {
+  pdfPageUrl: string;
+};
+
+const verifyBody = (body: any): RequestBody => {
+  const lesmisValidDomains = ['lesmis.la', 'www.lesmis.la'];
+  const { pdfPageUrl } = body;
+  if (!pdfPageUrl || typeof pdfPageUrl !== 'string') {
+    throw new Error(`'pdfPageUrl' should be provided in request body, got: ${pdfPageUrl}`);
+  }
+  const location = new URL(pdfPageUrl);
+  if (
+    !location.hostname.endsWith('.tupaia.org') &&
+    !lesmisValidDomains.includes(location.hostname) &&
+    !location.hostname.endsWith('localhost')
+  ) {
+    throw new Error(`'pdfPageUrl' is not valid, got: ${pdfPageUrl}`);
+  }
+  return { pdfPageUrl };
+};
+
+const buildParams = (req: any) => {
+  const cookies = cookie.parse(req.headers.cookie || '');
+  const { pdfPageUrl } = verifyBody(req.body);
+  const { host: apiDomain } = req.headers;
+  const location = new URL(pdfPageUrl);
+  const finalisedCookieObjects = Object.keys(cookies).map(name => ({
+    name,
+    domain: apiDomain,
+    url: location.origin,
+    httpOnly: true,
+    value: cookies[name],
+  }));
+  return { pdfPageUrl, cookies: finalisedCookieObjects };
+};
+
+export const exportToPDF = async (req: any) => {
```

Comment:
```suggestion
/**
 * @param pageUrl the url to visit and download as a pdf
 * @param userCookie the user's cookie to bypass auth, and ensure page renders under the correct user context
 * @returns pdf buffer
 */
export const downloadPageAsPdf = async (pageUrl: string, userCookie = '') => {
```

Might be nice to make the behaviour of this function a bit more clear, by just taking in the required parameters rather than a `req` object? Also makes it a bit more flexible in use 👍